source/extensions/filters/http/ext_proc/client_impl.cc

```diff
@@ -24,6 +24,13 @@ ExternalProcessorStreamPtr ExternalProcessorClientImpl::start(
                                              sidestream_watermark_callbacks);
 }
 
+void ExternalProcessorClientImpl::sendRequest(
+    envoy::service::ext_proc::v3::ProcessingRequest&& request, bool end_stream, const uint64_t,
+    RequestCallbacks*, StreamBase* stream) {
+  ExternalProcessorStream* grpc_stream = dynamic_cast<ExternalProcessorStream*>(stream);
```

Comment:
Please avoid the use of `dynamic_cast` This is indication that interfaces are not properly designed.

Reply:
For this dynamic_cast, passing ExternalProcessorStream* will cause circular dependency.   Thus a StreamBase* is passed in, which will be casted into ExternalProcessorStream* to send the message.

We had tried store an ExternalProcessorStream* pointer in the client_ object(https://github.com/envoyproxy/envoy/pull/36228). However, that does not work well and trigger a lot other complexities, so has to be abandoned.

So, can we take an exception here? We can discuss offline and have a follow up PR to clean this up after we find a solution.

Reply:
Added an TBD item under https://github.com/envoyproxy/envoy/issues/35488 to track this.